torch/csrc/jit/script/compilation_unit.h

```diff
@@ -186,12 +186,11 @@ struct TORCH_API CompilationUnit {
   }
 
   c10::TupleTypePtr get_named_tuple(const c10::QualifiedName& name) const {
-    for (const auto& cls : classes_) {
-      if (cls->name()->qualifiedName() == name.qualifiedName()) {
-        return cls->expect<TupleType>();
-      }
+    auto type = get_type(name);
+    if (!type) {
+      return nullptr;
     }
-    return nullptr;
+    return type->expect<TupleType>();
```

Comment:
This function seems suspicious because it can error if the type exists but is not a named tuple, but it returns nullptr (and does not error) if no type with the name is found.

Reply:
hmm yeah this follows the same logic as `get_class` or `get_interface`, and these methods are mainly used by compilation_unit api, do you think we should throw error in those cases if no type if found?